CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+  f (3 + 2) + 1
+
+  # good
+  f(3 + 2) + 1
+  ```
+
+* <a name="parens-as-args"></a>
+  If the first argument to a method begins with an open parenthesis, always
+  use parentheses in the method invocation. For example, write `f((3 + 2) + 1)`.
+<sup>[[link](#parens-as-args)]</sup>
+
+
+* <a name="no-nested-methods"></a>
+  Do not use nested method definitions, use lambda instead.
+  Nested method definitions actually produce methods in the same scope
+  (e.g. class) as the outer method. Furthermore, the "nested method" will be
+  redefined every time the method containing its definition is invoked.
+<sup>[[link](#no-nested-methods)]</sup>
+
+  ```Ruby
+  # bad
+  def foo(x)
+    def bar(y)
+      # body omitted
+    end
+
+    bar(x)
+  end
+
+  # good - the same as the previous, but no bar redefinition on every foo call
+  def bar(y)
+    # body omitted
+  end
+
+  def foo(x)
+    bar(x)
+  end
+
+  # also good
+  def foo(x)
+    bar = ->(y) { ... }
+    bar.call(x)
+  end
+  ```
+
+* <a name="lambda-multi-line"></a>
+  Use the new lambda literal syntax for single line body blocks. Use the
+  `lambda` method for multi-line blocks.
+<sup>[[link](#lambda-multi-line)]</sup>
+
+  ```Ruby
+  # bad
+  l = lambda { |a, b| a + b }
+  l.call(1, 2)
+
+  # correct, but looks extremely awkward
+  l = ->(a, b) do
+    tmp = a * 7
+    tmp * b / 50
+  end
+
+  # good
+  l = ->(a, b) { a + b }
+  l.call(1, 2)
+
+  l = lambda do |a, b|
+    tmp = a * 7
+    tmp * b / 50
+  end
+  ```
+
+* <a name="proc"></a>
+  Prefer `proc` over `Proc.new`.
+<sup>[[link](#proc)]</sup>
+
+  ```Ruby
+  # bad
+  p = Proc.new { |n| puts n }
+
+  # good
+  p = proc { |n| puts n }
+  ```
+
+* <a name="proc-call"></a>
+  Prefer `proc.call()` over `proc[]` or `proc.()` for both lambdas and procs.
+<sup>[[link](#proc-call)]</sup>
+
+  ```Ruby
+  # bad - looks similar to Enumeration access
+  l = ->(v) { puts v }
+  l[1]
+
+  # also bad - uncommon syntax
+  l = ->(v) { puts v }
+  l.(1)
+
+  # good
+  l = ->(v) { puts v }
+  l.call(1)
+  ```
+
+* <a name="underscore-unused-vars"></a>
+  Prefix with `_` unused block parameters and local variables. It's also
+  acceptable to use just `_` (although it's a bit less descriptive). This
+  convention is recognized by the Ruby interpreter and tools like RuboCop and
+  will suppress their unused variable warnings.
+<sup>[[link](#underscore-unused-vars)]</sup>
+
+  ```Ruby
+  # bad
+  result = hash.map { |k, v| v + 1 }
+
+  def something(x)
+    unused_var, used_var = something_else(x)
+    # ...
+  end
+
+  # good
+  result = hash.map { |_k, v| v + 1 }
+
+  def something(x)
+    _unused_var, used_var = something_else(x)
+    # ...
+  end
+
+  # good
+  result = hash.map { |_, v| v + 1 }
+
+  def something(x)
+    _, used_var = something_else(x)
+    # ...
+  end
+  ```
+
+* <a name="global-stdout"></a>
+  Use `$stdout/$stderr/$stdin` instead of `STDOUT/STDERR/STDIN`.
+  `STDOUT/STDERR/STDIN` are constants, and while you can actually reassign
```

Comment:
I mean, technically, you can do STDOUT.reopen(some_io), but I'll lean towards the globals here.

This shouldn't really affect Logstash anyway since we, in practice, should be avoiding std{out,err} in favor of logging. Exceptions are the stdout output and stdin input, and maybe some other small outliers.